### PR TITLE
scripts: add TESTCASE flag to run single unit test

### PR DESCRIPTION
**Pull request by wenjiaswe**
Enable user to run specified test case with different TIMEOUT expectation when needed.

### CODE REVIEW

File: test [hunk 1]
```diff
@@ -77,6 +97,12 @@ if [ "$GOARCH" == "amd64" ]; then
 	RACE="--race"
 fi
 
+if [ ! -z "${TESTCASE}" ]; then
+    RUN_ARG="-run=${TESTCASE}"
+else
+    RUN_ARG=""
+fi
+
```
Comment (hexfusion) [inline test]: maybe just cosmetic but saves an else statement.

```bash
RUN_ARG=""

if [ ! -z "${TESTCASE}" ]; then
     RUN_ARG="-run=${TESTCASE}"
fi
```

File: test [hunk 2]
```diff
@@ -57,6 +76,7 @@ else
 fi
 FMT=($FMT)
 
+
```
Comment (hexfusion) [inline test]: extra space?

File: test [hunk 3]
```diff
@@ -224,7 +290,17 @@ function cov_pass {
 
 function e2e_pass {
 	echo "Running e2e tests..."
-	go test -timeout 15m -v -cpu 1,2,4 "$@" "${REPO_PATH}/e2e"
+
+	# check if user provided time out, especially useful when just run one test case
```
Comment (hexfusion) [inline test]: indentation is a little off.

**Comment by wenjiaswe** [comment]
@hexfusion @jpbetz @gyuho Sorry, this is the same as #9378 , but I had the wrong rebase and branch for that one, so after I fixed that, github decides it should be closed. So I submitted this new one.

File: test (hunk 1 above)
Comment (wenjiaswe) [reply test]: Thanks, change applied.

File: test (hunk 2 above)
Comment (wenjiaswe) [reply test]: change applied.

File: test [hunk 4]
```diff
@@ -87,14 +111,54 @@ function unit_pass {
...
+
+    # check if user provided time out, especially useful when just run one test case
+    # expectation could be different
+    USERTIMEOUT=""
+    if [ -z "${TIMEOUT}" ]; then
+    	USERTIMEOUT="3m"
+    else
+    	USERTIMEOUT="${TIMEOUT}"
+    fi
+    go test ${GO_TEST_FLAG} -timeout "${USERTIMEOUT}"  "${COVER}" ${RACE} -cpu 1,2,4 ${RUN_ARG} "$@" "${TEST[@]}"
```
Comment (gyuho) [inline test]: Tabs from line 119? Indentation looks different than above lines.

File: test [hunk 5]
```diff
@@ -472,7 +545,7 @@ function commit_title_pass {
 		spaceCommas=$(echo "$commitMsg" | sed 's/ /\n/g' | grep -c ',$' || echo 0)
 		commaSpaces=$(echo "$commitMsg" | sed 's/,/\n/g' | grep -c '^ ' || echo 0)
 		if [[ $(echo "$commitMsg" | grep -c ":..*") == 0 || "$commitMsg" == "$pkgPrefix" || "$spaceCommas" != "$commaSpaces" ]]; then
-    			echo "$l"...
+				echo "$l"...
```
Comment (gyuho) [inline test]: revert this change?

Comment (wenjiaswe) [reply test]: OK, just applied. Thanks!
